opengever/maintenance/scripts/export_and_close_dossiers.py

```diff
@@ -155,12 +162,31 @@ def _get_output_path(self, basedir, name, ext='', i=0):
             return self._get_output_path(basedir, name, ext, i=i+1)
         return output_path
 
+    def _get_folder_path(self, dossier):
+        if not self.export_structure:
+            return self._get_output_path(self.output_directory, dossier.title)
+
+        if dossier.is_subdossier():
+            parent_path = self.dossier_path_mapping[dossier.absolute_url_path().rsplit("/", 1)[0]]
+        else:
+            segments = [el.id for el in dossier.aq_parent.aq_chain
+                        if IRepositoryFolder.providedBy(el)]
+            segments.append(self.output_directory)
+            parent_path = os.path.join(*reversed(segments))
+
+        path = self._get_output_path(parent_path, dossier.title)
+        self.dossier_path_mapping[dossier.absolute_url_path()] = path
+        return path
+
     def _export_dossier(self, dossier):
-        folder_path = self._get_output_path(self.output_directory, dossier.title)
-        os.mkdir(folder_path)
-        res = self.catalog.unrestrictedSearchResults(
-            path=dossier.absolute_url_path(),
-            object_provides=IBaseDocument.__identifier__)
+        folder_path = self._get_folder_path(dossier)
+        os.makedirs(folder_path)
+        if self.export_structure:
+            res = dossier.get_contained_documents()
```

Comment:
Does this handle documents inside tasks correctly?

Reply:
yes it does. It gets all documents in the dossier and then filters out the ones that are in a subdossier.